sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.throughputControl.ThroughputControlMode;
+import com.azure.cosmos.util.Beta;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+/**
+ * Group configuration which will be used in Throughput control.
+ */
+@Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class ThroughputControlGroup {
+
+    private final static boolean DEFAULT_USE_BY_DEFAULT = false;
+    private final static ThroughputControlMode DEFAULT_CONTROL_MODE = ThroughputControlMode.LOCAL;
+
+    private ThroughputControlMode controlMode;
+    private String groupName;
+    private String id;
+    private CosmosAsyncContainer targetContainer;
+    private Integer targetThroughput;
+    private Double targetThroughputThreshold;
+    private boolean useByDefault;
+
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup() {
```

Comment:
another option is that we instantiate ThroughputControlGroup through CosmosAsyncContainer and CosmosContainer that way you can bind the target container to the throughput control group.

```
ThroughputControlGroup group = cosmosAsyncContainer.createThroughputGroup(String groupName);
```

Reply:
Updated to use the proposed way.